opentracing-spring-cloud/src/test/java/io/opentracing/contrib/spring/cloud/hystrix/HystrixTraceCommandTest.java

```diff
@@ -0,0 +1,182 @@
+
+import com.netflix.hystrix.HystrixCommandGroupKey;
+import com.netflix.hystrix.HystrixCommandKey;
+import com.netflix.hystrix.contrib.javanica.annotation.HystrixCommand;
+import com.netflix.hystrix.strategy.HystrixPlugins;
+import io.opentracing.ActiveSpan;
+import io.opentracing.Span;
+import io.opentracing.contrib.spring.cloud.TestUtils;
+import io.opentracing.mock.MockSpan;
+import io.opentracing.mock.MockTracer;
+import io.opentracing.tag.Tags;
+import io.opentracing.util.ThreadLocalActiveSpanSource;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.autoconfigure.jmx.JmxAutoConfiguration;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.cloud.client.loadbalancer.LoadBalancerAutoConfiguration;
+import org.springframework.cloud.netflix.hystrix.EnableHystrix;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.EnableAspectJAutoProxy;
+import org.springframework.stereotype.Service;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+import static org.junit.Assert.assertEquals;
+
+@SpringBootTest(classes = {HystrixTraceCommandTest.TestConfig.class})
+@RunWith(SpringRunner.class)
+@DirtiesContext
+public class HystrixTraceCommandTest {
+
+    @EnableAspectJAutoProxy(proxyTargetClass = true)
+    @EnableHystrix
+    @Configuration
+    //This is required for circuit breaker tests to work correctly
+    @EnableAutoConfiguration(exclude = {LoadBalancerAutoConfiguration.class, JmxAutoConfiguration.class})
+    static class TestConfig {
+        @Bean
+        public MockTracer mockTracer() {
+            return new MockTracer(new ThreadLocalActiveSpanSource(), MockTracer.Propagator.TEXT_MAP);
+        }
+
+        @Bean
+        public GreetingService greetingService() {
+            return new GreetingService();
+        }
+    }
+
+    @Service
+    static class GreetingService {
+
+        @Autowired
+        MockTracer mockTracer;
+
+        @HystrixCommand
+        public String sayHello() {
+            Span span = mockTracer.buildSpan("sayHello")
+                    .startManual();
+
+            span.finish();
+            return "Hello!!!";
+        }
+
+        @HystrixCommand(fallbackMethod = "defaultGreeting")
+        public String sayHelloToUser(String user) {
+            Span span = mockTracer.buildSpan("sayHelloToUser")
```

Comment:
create it and finish it right away. that try block is redundant

Reply:
cleaned up, i was thinking that span.close is required i finally incase of exceptions et al